src/DepthImageToLaserScanROS.cpp

```diff
@@ -31,61 +31,84 @@
  * Author: Chad Rockey
  */
 
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <sensor_msgs/msg/laser_scan.hpp>
+
+#include <functional>
+
 #include <depthimage_to_laserscan/DepthImageToLaserScanROS.h>
 
+#define ROS_ERROR printf
+
 using namespace depthimage_to_laserscan;
-  
-DepthImageToLaserScanROS::DepthImageToLaserScanROS(ros::NodeHandle& n, ros::NodeHandle& pnh):pnh_(pnh), it_(n), srv_(pnh) {
-  boost::mutex::scoped_lock lock(connect_mutex_);
-  
-  // Dynamic Reconfigure
-  dynamic_reconfigure::Server<depthimage_to_laserscan::DepthConfig>::CallbackType f;
-  f = boost::bind(&DepthImageToLaserScanROS::reconfigureCb, this, _1, _2);
-  srv_.setCallback(f);
-  
-  // Lazy subscription to depth image topic
-  pub_ = n.advertise<sensor_msgs::LaserScan>("scan", 10, boost::bind(&DepthImageToLaserScanROS::connectCb, this, _1), boost::bind(&DepthImageToLaserScanROS::disconnectCb, this, _1));
+
+DepthImageToLaserScanROS::DepthImageToLaserScanROS(rclcpp::node::Node::SharedPtr & node):node_(node) {
+  rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_sensor_data;
+
+  custom_qos_profile.history = RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+  custom_qos_profile.depth = 50;
+  custom_qos_profile.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+  custom_qos_profile.durability = RMW_QOS_POLICY_DURABILITY_VOLATILE;
```

Comment:
Same question as https://github.com/ros2/ros_astra_camera/pull/17#discussion_r121443254 regarding performance improvement compared to the default sensor_data qos profile